src/rules/require-hook.ts

```diff
@@ -60,13 +75,35 @@ export default createRule({
       useHook: 'This should be done within a hook',
     },
     type: 'suggestion',
-    schema: [],
+    schema: [
+      {
+        type: 'object',
+        properties: {
+          allowedFunctionCalls: {
+            type: 'array',
+            items: { type: 'string' },
+          },
+        },
+        additionalProperties: false,
+      },
+    ],
   },
-  defaultOptions: [],
+  defaultOptions: [
+    {
+      allowedFunctionCalls: [],
+    },
+  ],
   create(context) {
+    const { allowedFunctionCalls } = context.options[0] ?? {};
+
     const checkBlockBody = (body: TSESTree.BlockStatement['body']) => {
       for (const statement of body) {
-        if (shouldBeInHook(statement)) {
+        if (
+          shouldBeInHook(
+            statement,
+            Array.isArray(allowedFunctionCalls) ? allowedFunctionCalls : [],
```

Comment:
This should always be an array due to the schema - if you're doing this because `allowedFunctionCalls` is being marked as possibly undefined by TS, you can provide a default value in the assignment